examples/chip-tool-darwin/templates/commands.zapt

```diff
@@ -60,22 +62,27 @@ public:
         params.{{asStructPropertyName label}} = [NSNumber numberWith{{asObjectiveCNumberType "" type false}}:m{{asUpperCamelCase label}}];
         {{/if_chip_complex}}
         {{/chip_cluster_command_arguments}}
-        [cluster {{asLowerCamelCase name}}With{{#if (hasArguments)}}Params:params completionHandler:{{else}}CompletionHandler:{{/if}}
-        {{#if hasSpecificResponse}}
-            ^(CHIP{{asUpperCamelCase clusterName}}Cluster{{asUpperCamelCase responseName}}Params * _Nullable values, NSError * _Nullable error) {
-                 NSLog(@"Values: %@", values);
-        {{else}}
-            ^(NSError * _Nullable error) {
-        {{/if}}
-            chipError = [CHIPError errorToCHIPErrorCode:error];
-            ChipLogProgress(chipTool, "Error: %s", chip::ErrorStr(chipError));
-            SetCommandExitStatus(chipError);
-            }];
+        uint16_t repeatCount = mRepeatCount.ValueOr(1);
+        while (repeatCount--)
+        {
+            [cluster {{asLowerCamelCase name}}With{{#if (hasArguments)}}Params:params completionHandler:{{else}}CompletionHandler:{{/if}}
+            {{#if hasSpecificResponse}}
+                ^(CHIP{{asUpperCamelCase clusterName}}Cluster{{asUpperCamelCase responseName}}Params * _Nullable values, NSError * _Nullable error) {
+                    NSLog(@"Values: %@", values);
+            {{else}}
+                ^(NSError * _Nullable error) {
+            {{/if}}
+                chipError = [CHIPError errorToCHIPErrorCode:error];
+                ChipLogProgress(chipTool, "Error: %s", chip::ErrorStr(chipError));
+                if (repeatCount == 0 || chipError != CHIP_NO_ERROR) {
+                    SetCommandExitStatus(chipError);
```

Comment:
As above, this does not have the right behavior on errors.

Reply:
Will update. Thank you.

Reply:
Fixed.